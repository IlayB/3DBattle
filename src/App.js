import RetroBackground from './components/retroBackground';

const App = () => {
  return (
    <div className="App">
      <RetroBackground />
    </div>
  );
};

export default App;
