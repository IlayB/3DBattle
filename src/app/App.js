import { BrowserRouter } from 'react-router-dom';
import { getToken } from 'firebase/messaging';

import { createTheme, ThemeProvider } from '@mui/material';

import RetroBackground from '../components/retroBackground';
import TopMenuNavigation from '../components/topMenuNavigation/TopMenuNavigation';
import { AppRoutes } from './route/AppRoutes';
import { messaging } from '../auth/firebaseApp';
import { useEffect } from 'react';

const MuiTheme = createTheme({
  typography: {
    fontFamily: 'New Rocker',
    allVariants: {
      color: 'white',
    },
    button: {
      textTransform: 'none',
    },
  },
  palette: {
    mode: 'dark',
  },
});

const App = () => {
  useEffect(() => {
    getToken(messaging, {
      vapidKey:
        'BPWRKU0cZTpZlMMK71z57nS-Bubi4NknfuIFC5NYpXVaPBfiPdA2BBOpyshiLqop-t4Em_amhl5X3vFBtI6dEvE',
    })
      .then((currentToken) => {
        if (currentToken) {
          console.log('TOKEN', currentToken);
        } else {
          console.log('No registration token available. Request permission to generate one.');
        }
      })
      .catch((err) => {
        console.log('An error occurred while retrieving token. ', err);
      });
  }, []);

  useEffect(() => {
    navigator.serviceWorker.addEventListener('message', ({ data }) =>
      console.log('GOT MESSAGE', data),
    );
  }, []);
  return (
    <div className="App">
      <ThemeProvider theme={MuiTheme}>
        <RetroBackground />
        <BrowserRouter basename={process.env.PUBLIC_URL}>
          <TopMenuNavigation />
          <AppRoutes />
        </BrowserRouter>
      </ThemeProvider>
    </div>
  );
};

export default App;
