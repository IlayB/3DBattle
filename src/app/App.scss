.App {
  margin: 0;
  height: 100%;
  background-color: red;
}