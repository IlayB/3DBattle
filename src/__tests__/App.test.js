import { render } from '@testing-library/react';

import App from '../App';

describe('App', () => {
  it('renders correctly', () => {
    const { baseElement } = render(<App />);
    expect(baseElement).toMatchSnapshot();
  });
});
