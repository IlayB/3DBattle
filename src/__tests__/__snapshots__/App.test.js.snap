// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<body>
  <div>
    <div
      class="App"
    >
      Hello
    </div>
  </div>
</body>
`;
